### PR TITLE
[#6412] Bump version number to 4.3.1. (main)

### PR DESCRIPTION
**Pull request by alanking**
In service of #6412

Cherry-picked from #7365 

I imagine this will be changed very soon due to #7361?

### CODE REVIEW

**Review by korydraughn — APPROVED** [review]
Pound it once @SwooshyCueb approves.

File: Doxyfile [hunk 1]
```diff
@@ -38,7 +38,7 @@ PROJECT_NAME           = "iRODS"
 # could be handy for archiving the generated documentation or if some version
 # control system is used.
 
-PROJECT_NUMBER         = main
+PROJECT_NUMBER         = 4.3.1
```
Comment (SwooshyCueb) [inline Doxyfile:41]: This should probably still say main? Not sure

Comment (trel) [reply Doxyfile:41]: yes please leave as main

Comment (korydraughn) [reply Doxyfile:41]: Ultimately this only matters when generating documentation.

I can see it going either way, but to keep things simple, let's set it to `main`.

Perhaps this file becomes a .in file and the project number is assigned at build time. Then we don't ever have to touch this file in relation to the version number again.

Comment (alanking) [reply Doxyfile:41]: Good catch.

Comment (alanking) [reply Doxyfile]: Created https://github.com/irods/irods/issues/7395 for the .in file idea. Resolving.

**Review by SwooshyCueb — APPROVED** [review]

**Pull request merged**